annotated_types/__init__.py

```diff
@@ -314,11 +315,20 @@ class Predicate(BaseMetadata):
     We do not specify what behaviour should be expected for predicates that raise
     an exception.  For example `Annotated[int, Predicate(str.isdigit)]` might silently
     skip invalid constraints, or statically raise an error; or it might try calling it
-    and then propogate or discard the resulting exception.
+    and then propagate or discard the resulting exception.
     """
 
     func: Callable[[Any], bool]
 
+    def __repr__(self) -> str:
+        if getattr(self.func, "__name__", "<lambda>") == "<lambda>":
+            return f"{self.__class__.__name__}({self.func!r})"
+        if isinstance(self.func, (types.MethodType, types.BuiltinMethodType)) and (
+            namespace := getattr(self.func.__self__, "__name__", None)
```

Comment:
Does this work on all Python versions we support now (3.8, so I think yes). Either way this is complex enough to warrant testing, is there a test that hits this or can you add one?

Reply:
Yep, it's extracted from a more-complex version in Hypothesis - I'm very confident that it works, and have added a test to make sure it _keeps_ working.